plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -72,15 +87,66 @@ rules :: Recorder (WithPriority Log) -> PluginId -> Rules ()
 rules recorder plId = do
   define (cmapWithPrio LogShake recorder) $
     \GetStanDiagnostics file -> do
-      config <- getPluginConfigAction plId
-      if pluginEnabledConfig plcDiagnosticsOn config then do
+      plugConfig <- getPluginConfigAction plId
+      if pluginEnabledConfig plcDiagnosticsOn plugConfig then do
           maybeHie <- getHieFile file
           case maybeHie of
             Nothing -> return ([], Nothing)
             Just hie -> do
-              let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
-              -- This should use Cabal config for extensions and Stan config for inspection preferences is the future
-              let analysis = runAnalysis Map.empty enabledInspections [] [hie]
+              let currentHSfromHIEAbs = hie_hs_file hie
+              currentHSfromHIERel <- liftIO $ makeRelativeToCurrentDirectory currentHSfromHIEAbs
+              -- This codes follows what 'runStan' does, from the module 'Stan'
+
+              -- There aren't any cli args. isLoud=False=Silent output
+              let isLoud = False -- Should this be enabled when debugging? Enables default stan cli output
+              let stanArgsConfigFile = Nothing -- There aren't any cli args
+
+              EnvVars{envVarsUseDefaultConfigFile} <- liftIO getEnvVars
+              logWith recorder Debug (LogDebug $
+                "envVarsUseDefaultConfigFile: " <> (T.pack $ show envVarsUseDefaultConfigFile))
+
+              let defConfTrial = envVarsUseDefaultConfigFile -- There aren't any cli args: <> stanArgsUseDefaultConfigFile
+              let useDefConfig = maybe True snd (trialToMaybe defConfTrial)
+
+              tomlConfig <- liftIO $ getTomlConfig isLoud useDefConfig stanArgsConfigFile
+              let configTrial = finaliseConfig $ defaultConfig <> tomlConfig -- There aren't any cli args: <> stanArgsConfig
+              logWith recorder Debug (LogDebug $ "Final stan config result\n" <> ( T.pack $ show configTrial))
```

Comment:
Since we are using the colog style logger, prefer a custom constructor, for example `LogStanConfigSearchResult configTrial`.